### PR TITLE
LibTorchSegmentation static lib 

### PR DESCRIPTION
**Pull request by leonelhs**
First of all, tanks for share your awesome job!

For learn and study purposes I have been playing with this library. 
In order to increase the build speed I have splitted the project in two parts,  one is a core static library and the second is a test-demo project. CMake will install the core lib in the /usr/local/lib in order to avoid compile the whole project for every change in the test-demo project.

Due to I will keep exploring the wonderful possibilities of this project , I would like to keep the original repository as reference.

### CODE REVIEW

**Comment by AllentDan** [comment]
Thanks for the great contribution. I will review it ASAP.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,35 +1,148 @@
...
+    "src/architectures/FPNDecoder.h"
+    "src/architectures/FPN.h"
+    "src/architectures/LinknetDecoder.h"
+    "src/architectures/LinkNet.h"
+    "src/architectures/PANDecoder.h"
+    "src/architectures/PAN.h"
+    "src/architectures/PSPNetDecoder.h"
+    "src/architectures/PSPNet.h"
+    "src/architectures/UNetDecoder.h"
+    "src/architectures/UNet.h"
```
Comment (AllentDan) [inline CMakeLists.txt]: Could you make it like `*.h` in `src/architectures`?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,35 +1,148 @@
...
+    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/segmentation/backbones)
+
+set(UTILS_PUBLIC_HEADERS
+    "src/utils/Augmentations.h"
+    "src/utils/_dirent.h"
+    "src/utils/InterFace.h"
+    "src/utils/loss.h"
+    "src/utils/readfile.h"
+    "src/utils/util.h"
+    "src/utils/json.hpp"
```
Comment (AllentDan) [inline CMakeLists.txt]: the same as above. `*.h` was a prefer.

**Review by AllentDan — COMMENTED** [review]
Is it possible that we give an option for user to control the compiling process so that the user can either build static lib or dynamic lib?

**Comment by leonelhs** [comment]
Its done. After googling a little bit I have achieved to group all *.h files in a single sentence, also choose to build a shared or static lib its done.

**Pull request merged**